cmake/KokkosCTest.cmake.in

```diff
@@ -0,0 +1,243 @@
+
+# ---------------------------------------------------------------------------- #
+# -- Settings
+# ---------------------------------------------------------------------------- #
+## -- Process timeout in seconds
+set(CTEST_TIMEOUT           "7200")
+## -- Set output to English
+set(ENV{LC_MESSAGES}        "en_EN" )
+
+
+# ---------------------------------------------------------------------------- #
+# -- Copy ctest configuration file
+# ---------------------------------------------------------------------------- #
+macro(COPY_CTEST_CONFIG_FILES)
+
+    foreach(_FILE CTestConfig.cmake CTestCustom.cmake)
+
+        # if current directory is not binary or source directory
+        if(NOT "${CMAKE_CURRENT_LIST_DIR}" STREQUAL "${CTEST_BINARY_DIRECTORY}" AND
+           NOT "${CTEST_SOURCE_DIRECTORY}" STREQUAL "${CTEST_BINARY_DIRECTORY}")
+
+            # if file exists in current directory
+            if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/${_FILE})
+                configure_file(${CMAKE_CURRENT_LIST_DIR}/${_FILE}
+                    ${CTEST_BINARY_DIRECTORY}/${_FILE} COPYONLY)
+            endif()
+
+        # if source and binary differ
+        elseif(NOT "${CTEST_SOURCE_DIRECTORY}" STREQUAL "${CTEST_BINARY_DIRECTORY}")
+
+            # if file exists in source directory but not in binary directory
+            if(EXISTS ${CTEST_SOURCE_DIRECTORY}/${_FILE} AND
+               NOT EXISTS ${CTEST_BINARY_DIRECTORY}/${_FILE})
+                configure_file(${CTEST_SOURCE_DIRECTORY}/${_FILE}
+                    ${CTEST_BINARY_DIRECTORY}/${_FILE} COPYONLY)
+            endif()
+
+        endif()
+    endforeach()
+
+endmacro()
+
+ctest_read_custom_files("${CMAKE_CURRENT_LIST_DIR}")
+
+message(STATUS "CTEST_MODEL: ${CTEST_MODEL}")
+
+#-------------------------------------------------------------------------#
+# Start
+#
+message(STATUS "")
+message(STATUS "[${CTEST_BUILD_NAME}] Running START_CTEST stage...")
+message(STATUS "")
+
+ctest_start(${CTEST_MODEL} TRACK ${CTEST_MODEL} ${APPEND_CTEST}
+    ${CTEST_SOURCE_DIRECTORY} ${CTEST_BINARY_DIRECTORY})
+
+
+#-------------------------------------------------------------------------#
+# Config
+#
+copy_ctest_config_files()
+ctest_read_custom_files("${CTEST_BINARY_DIRECTORY}")
+
+
+#-------------------------------------------------------------------------#
+# Update
+#
+message(STATUS "")
+message(STATUS "[${CTEST_BUILD_NAME}] Running CTEST_UPDATE stage...")
+message(STATUS "")
+
+ctest_update(SOURCE "${CTEST_SOURCE_DIRECTORY}"
+    RETURN_VALUE up_ret)
+
+
+#-------------------------------------------------------------------------#
+# Configure
+#
+message(STATUS "")
+message(STATUS "[${CTEST_BUILD_NAME}] Running CTEST_CONFIGURE stage...")
+message(STATUS "")
+
+ctest_configure(BUILD "${CTEST_BINARY_DIRECTORY}"
+    SOURCE ${CTEST_SOURCE_DIRECTORY}
+    ${APPEND_CTEST}
+    OPTIONS "${CTEST_CONFIGURE_OPTIONS}"
+    RETURN_VALUE config_ret)
```

Comment:
Does that mean that the build will attempt to build even when configuration failed?

Reply:
Yeah but it immediately fails so there's no harm